java/code/src/com/redhat/rhn/frontend/action/configuration/files/ManageRevisionSubmit.java

```diff
@@ -303,6 +306,7 @@ public int deleteRevision(RhnSetElement elementIn, User userIn) {
     private void checkAcl(RequestContext requestContext) {
         //Throws an exception if the user does not have permission to edit the channel.
         User user = requestContext.getCurrentUser();
+        ensureRoleBasedAccess(user, "config.files", Namespace.AccessMode.W);
```

Comment:
Shouldn't this be done on the general web filter? Or this is one of those cases that reading and writing are mixed together?

Reply:
Exactly, only upload and delete actions require W mode.